common/dbconfig.cpp

```diff
@@ -0,0 +1,103 @@
+// Copyright 2021-present Pinterest. All Rights Reserved.
+//
+// @author Premkumar (prem@pinterest.com)
+//
+
+#include "common/dbconfig.h"
+#include "common/segment_utils.h"
+#include <utility>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+
+DECLARE_int32(replicator_replication_mode);
+
+namespace common {
+
+DBConfigManager* DBConfigManager::get() {
+  static DBConfigManager instance;
+  return &instance;
+}
+
+DBConfigManager::DBConfigManager() : dbConfig(new DBConfig()) {
```

Comment:
C++ question: `new DBConfig()`, or shall we use `std::make_shared<DBConfig>()`

Reply:
`std::make_shared<DBConfig>()` would be typically cleaner and should be used in normal init. This being in the initializer list of ctor, `dbConfig(new DBConfig())` is shorter ..